src/ops/project.rs

```diff
@@ -34,6 +93,25 @@ impl Project {
             self.emit.as_ref().map_or(col, |emit| emit[col])
         }
     }
+
+    fn eval_expression(&self, expression: &ProjectExpression, record: &Record) -> DataType {
+        let left = match expression.left {
+            ProjectExpressionBase::Column(i) => &record[i],
+            ProjectExpressionBase::Literal(ref data) => data,
+        }.clone();
```

Comment:
I don't think the `.clone()` here (or below) should be necessary?

Reply:
Is there any way to do this without overloading the arithmetic operators on `&DataType` instead of `DataType`? Pushed a commit [here](https://github.com/ekmartin/distributary/commit/d1c65298f72a04a9e86ee6966a66f70d3f7f334d0) for how that would look.

Tried playing around with ownership stuff, but couldn't figure out how to do it without having to `.clone()` the `ProjectExpression` passed into `eval_expression` and the `DataType` at `record[i]`.

Reply:
I doing arithmetic on `&DataType` is fine. That said, the problem here is less about cloning the record but about the fact that the whole `ProjectExpressions` is being cloned (which is larger and more complex). Strictly speaking, _only_ `record[i]` needs to be cloned, correct? Then you could use the arithmetic operators directly on `DataType` and `Literal`, without having to overload them for `ProjectExpression`.

Reply:
> Strictly speaking, only `record[i]` needs to be cloned, correct? Then you could use the arithmetic operators directly on DataType and Literal, without having to overload them for ProjectExpression.

I might be misunderstanding, but I think that's what I'm doing? `left` and `right` are `DataType` objects here, and the operators are overloaded [directly](https://github.com/mit-pdos/distributary/pull/30/files#diff-712c61e60613c53cdcc824c13afd1800R269) on `DataType`.

Reply:
Oh, yes, you're right, I misread.